util/Logging.py

```diff
@@ -21,54 +31,60 @@ def limit(_limit_level: int) -> callable:
     Levels are inclusive, so if you are level 2, you cannot use level 1 commands either.
     """
 
-    def decorator(func: callable) -> callable:
+    def decorator(
+        func: Callable[LimitedCommandParams, Awaitable[LimitedCommandReturnValue]],
+    ) -> Callable[LimitedCommandParams, Awaitable[LimitedCommandReturnValue | None]]:
         @functools.wraps(func)
-        async def wrapper(*args, **kwargs) -> callable:
+        async def wrapper(
+            *args: LimitedCommandParams.args,
+            **kwargs: LimitedCommandParams.kwargs,
+        ) -> LimitedCommandReturnValue | None:
             """
             Wrapper to determine if the user is limited or not.
             """
 
-            # Get the author id
-            author_id = None
+            ctx = None
             for arg in args:
-                try:
-                    author_id = arg.author.id
+                if isinstance(arg, discord.ApplicationContext):
                     ctx = arg
-                except AttributeError:  # Type is not a context
-                    continue
-
-            if (
-                author_id is not None
-            ):  # if no author exists, then we cannot limit, return func instead
-                conn = sqlite3.connect("util/database.sqlite")
-                c = conn.cursor()
-                limit_level = c.execute(
-                    "SELECT limitLevel from user where uid = ?",
-                    (author_id,),
-                ).fetchone()
-
-                if limit_level is None:  # User DOES NOT exist in database, add them.
-                    c.execute(
-                        "INSERT INTO user VALUES (?, ?, ?, ?, ?, ?)",
-                        (author_id, 0, False, None, 0, None),
-                    )  # See ERD.mdj
-                    limit_level = 0
-                    conn.commit()
-                else:
-                    limit_level = limit_level[0] or 0
-
-                if limit_level >= _limit_level:
-                    embed = EmbedBuilder(
-                        title="You cannot use this command!",
-                        description="You are limited from using this command! Please contact a moderator if you believe this is a mistake.",
-                        color=0xFF0000,
-                    ).build()
-                    await ctx.respond(embed=embed, ephemeral=True)
-                    Log(
-                        f"$ tried to use {ctx.command.name}. But is limited from using it.",
-                        ctx.author,
-                    )
-                    return None
+                    break
+
+            if not ctx or not ctx.author or not ctx.author.id:
+                # in order to actually do any limiting, we need to know who used the command
+                Log(
+                    f"@limit({limit_level_requirement}) is BROKEN for {func!r}. Failed to find an author!",
```

Comment:
Limit isn't broken, someone just added the decorator to a function that isn't a discord command

Reply:
That's true, the log line is inaccurate. I would still suggest adding a log like "@limit() was used incorrectly" or something like that so we can be sure to fix the bad usage. That's definitely out-of-scope for this PR though, so I've removed it.